src/Mod/Part/App/ExtrusionHelper.cpp

```diff
@@ -156,13 +157,10 @@ void ExtrusionHelper::makeDraft(const TopoDS_Shape& shape,
                     createTaperedPrismOffset(TopoDS::Wire(singleWire), vecRev, distanceRev, true, offsetWire);
                 }
                 else {
-                    // there is an OCC bug with single-edge wires (circles), see inside createTaperedPrismOffset
-                    if (numEdges > 1 || !isPartDesign)
-                        // inner wires must get the negated offset
-                        createTaperedPrismOffset(TopoDS::Wire(singleWire), vecRev, -distanceRev, true, offsetWire);
-                    else
-                        // circles in PartDesign must not get the negated offset

```

Comment:
are you sure this should be removed? don't we need an occt version check here? @donovaly thoughts on this?

Reply:
It works without. But I may not have cleaned enough.
Furthermore, I saw the issue #8587 and I am looking into it as it relates also to taper errors.

Reply:
(I did not correct the Part WB because I don't use it much. I did not implement the checkbox in Part and the discrepency of inner taper angle is still there.)

Reply:
If you look at the comment it is saying this was done to workaround an occt bug, if this is not necessary anymore it likely means it was solved in occt, but we should identify in which version and then apply the workaround for the versions where it's needed

Reply:
Looking again, this file is used in PartWB too, you can't modify it without taking into account it's use in Part WB

Reply:
For the PartWB, the checkbox implementation is needed.
For occt, I am looking into it for the #8587. Do you know how to display occt objects in the console? Is there a library for that? where is it?

Reply:
> For the PartWB, the checkbox implementation is needed.

As I wrote we purposely don't have this option. To avoid that we open this discussion here, let's focus with this PR on the actual bug that OCC treats circles (single-line wire) in a different way than polyline wires .

>Do you know how to display occt objects in the console?

What do you mean? OCC (OpenCasCade) od the CAD library FreeCAD is based on. FreeCAD's Part and PartDesign WB are basically a frontend for OCC.

Reply:
I found out how to do that. For OCC objects, DumpJson() is the equivalent to ToString(). To output it to the console, I use         `
myOcctObject.DumpJson(aSStream);
std::string& resultstr = aSStream.str();
Base::Console().Message(resultstr.c_str());
`
My question was if there was librairie/functions in Freecad taylored to do something like that to ease the debug of OCC objects.

However, this method is not convenient. So I tried to compile Freecad in debug mode and use the occt.natvis visualizer for Visual Studio.

But now, I struggled to compile the OCC in debug and use this debug version in the Freecad debug version.
The cmake points to libpack occ version.

It would be great if the OCC debug version could be included in the libpack and a switch be implemented in the cmake to be able to compile and use both freecad and occ in debug mode.

Reply:
Ok, it seems it has the debug version of OCC. 
However I saw the thread: https://forum.freecad.org/viewtopic.php?f=10&t=47017, which does not mention that.
I'll verify.